tests/test_qaoa.py

```diff
@@ -660,10 +671,29 @@ def circuit(params, **kwargs):
             qml.layer(qaoa_layer, 2, params[0], params[1])
 
         # Defines the device and the QAOA cost function
-        dev = qml.device('default.qubit', wires=len(wires))
+        dev = qml.device("default.qubit", wires=len(wires))
         cost_function = qml.ExpvalCost(circuit, cost_h, dev)
 
         res = cost_function([[1, 1], [1, 1]])
         expected = -1.8260274380964299
 
         assert np.allclose(res, expected, atol=tol, rtol=0)
+
+
+class TestCycles:
+
+    """Test `cycles` functions are behaving correctly"""
+
+    def test_edges_to_wires(self):
+        """Test that map_edges_to_wires returns the correct mapping"""
+        g = nx.lollipop_graph(4, 1)
+        r = edges_to_wires(g)
+
+        assert r == {(0, 1): 0, (0, 2): 1, (0, 3): 2, (1, 2): 3, (1, 3): 4, (2, 3): 5, (3, 4): 6}
+
+    def test_wires_to_edges(self):
+        """Test that map_wires_to_edges returns the correct mapping"""
+        g = nx.lollipop_graph(4, 1)
+        r = wires_to_edges(g)
+
+        assert r == {0: (0, 1), 1: (0, 2), 2: (0, 3), 3: (1, 2), 4: (1, 3), 5: (2, 3), 6: (3, 4)}
```

Comment:
What do you think about having these tests in their own test file? We might be adding quite a few for the `cycle` problem.

Reply:
Yes there will be a lot of tests here so would be good to move this into a separate  file, this would exist at the same level as the rest of qaoa. Wondering if this would be better to do when we revisit documentation structure so we have an idea of the full structure?